### PR TITLE
Add documentation for building programs using AssemblyScript.

### PR DESCRIPTION
**Pull request by sunfishcode**
Thanks to @dcodeIO and @torch2424 for their help answering my questions!

### CODE REVIEW

File: docs/wasm-assemblyscript.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+To create a program which can be run directly as a command, pass `--runtime half`
+to the AssemblyScript linker. This selects the [half runtime], which ensures that
+the generated wasm module doesn't contain any extraneous exports. (This isn't
+strictly required today, but the handling of extraneous exports may change in
+the future, so it's encouraged. As a bonus, it also reduces code size.)
+
+To create a program which can be loaded as a library and used from other modules,
+no special options are needed.
+
+[Here is a repository containing an example Hello World program][WASI hello world]
```
Comment (fitzgen) [inline docs/wasm-assemblyscript.md:15]: Might be worth inlining the "hello, world" example, as we do for the Rust docs.

Comment (torch2424) [reply docs/wasm-assemblyscript.md:15]: Yoooo Whaddup @fitzgen ! :smile: 

Yeah I agree, totally inlining this here would be a good move. 

I would add as:

---

For example, to compile an AssemblyScript module with WASI, you could write the following AssemblyScript code:

**hello-world.ts**
```
import "wasi";
import {Console} from "as-wasi";

Console.log('Hello World!\n');
```

Which uses [as-wasi](https://github.com/jedisct1/as-wasi) as a dependency to make working withe AssemblyScript WASI bindings easier. Then, you can run:

`asc hello-world.ts -b hello-world.wasm`

To get a wasm module that will print "Hello World!" to stdout.

---

Or something along those lines :smile:

Comment (sunfishcode) [reply docs/wasm-assemblyscript.md]: Thanks! I've now copied the content inline and covered it in the doc.

File: examples/assemblyscript/package.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
+{
+  "name": "wasi-hello-world",
+  "version": "1.0.0",
+  "description": "Hello world in Wasi with AS and as-wasi",
+  "main": "index.js",
+  "scripts": {
+    "build": "asc wasi-hello-world.ts -b wasi-hello-world.wasm -t wasi-hello-world.wat --runtime half",
+    "wasmtime": "wasmtime wasi-hello-world.wasm"
+  },
+  "author": "Aaron Turner",
```
Comment (torch2424) [inline examples/assemblyscript/package.json]: Just wanted to make sure your cool with the license and me being the author. Feel free to change to whichever you think is best for the wasmtime repo :smile:

Comment (sunfishcode) [reply examples/assemblyscript/package.json]: Thanks for the heads up! I think we can leave it as is for now.

Of course, if you'd ever like us to change it, feel free to reach out.

**Pull request merged**